types/musickit-js/MusicKit.API.d.ts

```diff
@@ -1,20 +1,1531 @@
 declare namespace MusicKit {
     /**
-     * @todo add types for Apple Music API Objects
+     * A resource—such as an album, song, or playlist.
+     * https://developer.apple.com/documentation/applemusicapi/resource
      */
     interface Resource {
-        attributes: {
+        id: string;
+        // (Required) Persistent identifier of the resource.
+        type: string;
+        // (Required) The type of resource.
+        href?: string;
+        // A URL subpath that fetches the resource as the primary object. This member is only present in responses.
+        attributes?: {
+            [key: string]: any;
+        };
+        // Attributes belonging to the resource (can be a subset of the attributes). The members are the names of the attributes defined in the object model.
+        relationships?: {
```

Comment:
```ts
interface Something {
{
	href?: string;
	// A URL subpath that fetches the relationship resources as the primary object. This member is only present in responses.
	next?: string;
	// Link to the next page of resources in the relationship. Contains the offset query parameter that specifies the next page. See Fetch Resources by Page.
	data: Resource[];
	// (Required) One or more destination objects.
	meta?: Record<string, any>;
	// Contextual information about the relationship for the request or response.
}
```


In order to simplify this type:

```suggestion
        relationships?: Record<string, Something | Something[]>
```

Reply:
Also, I'm a bit more used to see comments above code, like this:

```ts
interface Relationship {
{
	// A URL subpath that fetches the relationship resources as the primary object. This member is only present in responses.
	href?: string;
	// Link to the next page of resources in the relationship. Contains the offset query parameter that specifies the next page. See Fetch Resources by Page.
	next?: string;
	// (Required) One or more destination objects.
	data: Resource[];
	// Contextual information about the relationship for the request or response.
	meta?: Record<string, any>;
}
```

Reply:
Thank you for your accurate review!
We have improved the redundant code by abstracting it and would appreciate your re-review.